src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NetCorePackageReferenceProject.cs

```diff
@@ -275,11 +276,26 @@ private static IEnumerable<PackageReference> GetPackageReferences(IEnumerable<Li
                         originalFramework = framework.GetShortFolderName();
                     }
 
-                    await conditionalService.AddAsync(
+                    var reference = await conditionalService.AddAsync(
                         packageId,
                         formattedRange,
                         TargetFrameworkCondition,
                         originalFramework);
+
+                    // SuppressParent could be set to All if developmentDependency flag is true in package nuspec file.
+                    if (installationContext.SuppressParent != LibraryIncludeFlagUtils.DefaultSuppressParent &&
+                        installationContext.IncludeType != LibraryIncludeFlags.All)
```

Comment:
why is IncludeType != LibraryIncludeFlags.All being checked here?

Reply:
Because we only want to set these `PrivateAssets` and `Include` flags when they don't have default values which means either customer overrode these in his project or package defined `developmentDependency` flag in its nuspec file.